### PR TITLE
Add FIM capabilities to cookbook

### PR DESCRIPTION
**Pull request by ch7ck**
Todo:

Update README to include information from comments

### CODE REVIEW

**Comment by mihasya** [comment]
(don't merge this stuff until I'm done with the agent changes please)

File: .kitchen.yml [hunk 1]
```diff
@@ -12,10 +12,12 @@ platforms:
...
   - name: default
     run_list:
-      - recipe[opsmatic::agent]
+     - recipe[opsmatic::agent]
+      
     attributes:
       {
         "opsmatic": {
-          "integration_token": "4f15f0b3-6881-41e0-a8a7-c1d9e85c8018"
+          "integration_token": "<YOUR INTEGRATION TOKEN>",
```
Comment (mihasya) [inline .kitchen.yml]: the integration token in .kitchen.yaml should just be the token for our automated tests org. it's the test token, it's fine.

File: metadata.rb [hunk 2]
```diff
@@ -1,10 +1,10 @@
 name             'opsmatic'
-maintainer       'Marcus Barczak'
+maintainer       'Joe Friedrich'
```
Comment (mihasya) [inline metadata.rb]: Go ahead and just set this to `Opsmatic, Inc`

File: recipes/agent.rb [hunk 3]
```diff
@@ -30,8 +30,10 @@
   not_if { ::File.exists?("/var/db/opsmatic-agent/data")}
 end
 
+include_recipe "opsmatic::file-integrity-monitoring"
+
 # configure the service
 service "opsmatic-agent" do
   provider Chef::Provider::Service::Upstart
-  action [:enable, :start]
+  action [:enable, :start, :reload]
```
Comment (mihasya) [inline recipes/agent.rb]: why are we doing a reload here?

File: recipes/common.rb [hunk 4]
```diff
@@ -6,3 +6,13 @@
   group "root"
   mode "00644"
 end
+
+# Config File for Opsmatic Agent
+# https://opsmatic.com/app/docs/agent-configuration
+
+template "/etc/opsmatic-agent.conf" do
```
Comment (mihasya) [inline recipes/common.rb]: this should be in `agent.rb`

File: templates/default/opsmatic-agent.conf.erb [hunk 5]
```diff
@@ -0,0 +1,12 @@
+# automatically generated by chef cookbook opsmatic-cookbook
+# additional info -> https://opsmatic.com/app/docs/agent-configuration
+
+<% if node["opsmatic"]["host_alias"] && !node["opsmatic"]["host_alias"].empty? %>
+# Host Alias
+host_alias = "<%= node["opsmatic"]["host_alias"] %>"
+<% end %>
+<% if node["opsmatic"]["groups"] && !node["opsmatic"]["groups"].empty? %>
```
Comment (mihasya) [inline templates/default/opsmatic-agent.conf.erb]: the attribute itself should be a list, and we should join that list on a comma in the template itself.

**Comment by ch7ck** [comment]
ok I addressed everything.

**Comment by mihasya** [comment]
Cool, I'll look again once I'm done with the agent and warlock changes required to make this work.

**Comment by mihasya** [comment]
:+1:

File: templates/default/opsmatic-agent.conf.erb [hunk 6]
```diff
@@ -0,0 +1,11 @@
...
+# Group(s) that include this host_alias, comma separated
```
Comment (mihasya) [inline templates/default/opsmatic-agent.conf.erb]: Should be "groups of which this host is a member," `host_alias` unrelated here

File: templates/default/opsmatic-agent.conf.erb [hunk 7]
```diff
@@ -0,0 +1,11 @@
+# automatically generated by chef cookbook opsmatic-cookbook
+# additional info -> https://opsmatic.com/app/docs/agent-configuration
+
+<% if node["opsmatic"]["host_alias"] && !node["opsmatic"]["host_alias"].empty? %>
+# Host Alias
```
Comment (mihasya) [inline templates/default/opsmatic-agent.conf.erb]: This should say something a little more helpful, or just not exist

**Pull request merged**